aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -460,11 +511,16 @@ module AptosFramework::Stake {
         pool_address: address,
         owner_cap: &OwnerCapability,
         coins: Coin<TestCoin>,
-    ) acquires StakePool, StakePoolEvents, ValidatorSet, ValidatorSetConfiguration {
+    ) acquires StakePool, StakePoolEvents, ValidatorSet, ValidatorSetConfiguration, ValidatorSetVotingPower {
         assert!(owner_cap.pool_address == pool_address, Errors::invalid_argument(ENOT_OWNER));
+        let amount_added = Coin::value<TestCoin>(&coins);
+
+        let validator_set_config = borrow_global<ValidatorSetConfiguration>(@CoreResources);
+        let validator_set_voting_power = borrow_global_mut<ValidatorSetVotingPower>(@CoreResources);
+        validator_set_voting_power.joining = validator_set_voting_power.joining + (amount_added as u128);
+        validate_voting_power_increase(validator_set_voting_power, validator_set_config);
```

Comment:
this doesn't look right, we should only count the pools that are active or pending_active? if it's inactive, it seems double add to the joining field when the pool actually joins

Reply:
Amazing catch!!! I updated the logic according to the suggestion and added unit tests to make sure we don't count inactive validator's stake in ValidatorSetVotingPower#joining.